templates/CRM/Contribute/Form/Contribution/Main.tpl

```diff
@@ -342,16 +316,22 @@
   }
 
   cj('input[id="is_recur"]').on('change', function() {
-    showRecurHelp();
+    toggleRecur();
   });
 
-  function showRecurHelp( ) {
-    var showHelp = cj('input[id="is_recur"]:checked');
-    if ( showHelp.val() > 0) {
+  function toggleRecur( ) {
+    var isRecur = cj('input[id="is_recur"]:checked');
+    var allowAutoRenew = {/literal}'{$allowAutoRenewMembership}'{literal};
+    if ( allowAutoRenew && cj("#auto_renew") ) {
```

Comment:
@monishdeb I think moving this showHideAutoRenew call up here (out of the later if) solves the regression problem you found. showHideAutoRenew is clever enough to ... show AND hide